libcaf_core/caf/detail/type_traits.hpp

```diff
@@ -607,6 +607,12 @@ constexpr bool can_insert_elements() {
   return can_insert_elements_impl<T>(static_cast<T*>(nullptr));
 }
 
+template <class Test, template <class...> class Ref>
+struct is_specialization : std::false_type { };
+
+template <template <class...> class Ref, class... Args>
+struct is_specialization<Ref<Args...>, Ref> : std::true_type { };
+
 } // namespace detail
```

Comment:
Having the template template argument as second argument seems a bit counterintuitive to me. Did you consider doing it this way?

```cpp
/// -- add docu here ---
template <template <class...> class Tpl, class T>
struct is_specialization : std::false_type { };

template <template <class...> class T, class... Ts>
struct is_specialization<T, T<Ts...>> : std::true_type { };
```